plugin/PDFDocumentPage.qml

```diff
@@ -189,9 +375,182 @@ DocumentPage {
         }
     }
 
+    ContextMenu {
+        id: contextMenuText
+        property variant annotation
+        property point at
+
+        MenuItem {
+            visible: contextMenuText.annotation === undefined
+                     || contextMenuText.annotation === null
+            //% "Add note"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-add")
+            onClicked: {
+                var annotation = textComponent.createObject(contextMenuText)
+                annotation.color = "#202020"
+                pdfDocument.create(annotation,
+                                   function() {
+                                       var at = view.getPositionAt(contextMenuText.at)
+                                       annotation.attachAt(pdfDocument, at[0], at[2], at[1])
+                                   })
+            }
+        }
+        MenuItem {
+            visible: contextMenuText.annotation !== undefined
+                     && contextMenuText.annotation !== null
+            //% "Edit"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-edit")
+            onClicked: pdfDocument.edit(contextMenuText.annotation)
+        }
+        MenuItem {
+            visible: contextMenuText.annotation !== undefined
+                     && contextMenuText.annotation !== null
+            //% "Delete"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-clear")
+            onClicked: contextMenuText.annotation.remove()
+        }
+        Component {
+            id: textComponent
+            PDF.TextAnnotation { }
+        }
+    }
+
+    ContextMenu {
+        id: contextMenuHighlight
+        property variant annotation
+
+        InfoLabel {
+            id: infoContents
+            visible: infoContents.text != ""
+            width: parent.width
+            anchors.topMargin: Theme.paddingSmall
+            height: implicitHeight + 2 * Theme.paddingSmall
+            font.pixelSize: Theme.fontSizeSmall
+            wrapMode: Text.Wrap
+            elide: Text.ElideRight
+            maximumLineCount: 2
+            color: Theme.highlightColor
+            opacity: .6
+            text: {
+                if (contextMenuHighlight.annotation !== null
+                    && contextMenuHighlight.annotation !== undefined
+                    && contextMenuHighlight.annotation.contents != "") {
+                    return (contextMenuHighlight.annotation.author != ""
+                            ? "(" + contextMenuHighlight.annotation.author + ") " : "")
+                    + contextMenuHighlight.annotation.contents
+                } else {
+                    return ""
+                }
+            }
+        }
+        Row {
+            height: Theme.itemSizeExtraSmall
+            Repeater {
+                id: colors
+                model: ["#db431c", "#ffff00", "#8afa72", "#00ffff",
+                "#3828f9", "#a328c7", "#ffffff", "#989898",
+                "#000000"]
```

Comment:
Corrected. Normally, I do like that, but maybe after several code moves Emacs decided differently and I didn't correct it afterwards.